easybuild/easyblocks/w/wxpython.py

```diff
@@ -41,41 +43,106 @@ class EB_wxPython(PythonPackage):
+            wxflag = ''
+            if get_software_root('wxWidgets'):
+                wxflag = '--use_syswx'
+            script = 'build.py'
+            cmd = INSTALL_CMD % {
+                'preinst_opts': preinst_opts,
+                'pycmd': self.python_cmd,
+                'script': script,
+                'prefix': self.installdir,
+            }
+            cmd = cmd + " %s -v install" % wxflag
+        else:
+            script = os.path.join('wxPython', 'build-wxpython.py')
+            cmd = INSTALL_CMD % {
+                'preinst_opts': preinst_opts,
+                'pycmd': self.python_cmd,
+                'script': script,
+                'prefix': self.installdir,
+            }
+            cmd = cmd + " --wxpy_installdir=%s --install" % self.installdir
+
         run_cmd(cmd, log_all=True, simple=True)
 
     def sanity_check_step(self):
         """Custom sanity check for wxPython."""
         majver = '.'.join(self.version.split('.')[:2])
         shlib_ext = get_shared_lib_ext()
-        py_bins = ['alacarte', 'alamode', 'crust', 'shell', 'wrap', 'wxrc']
+        py_bins = ['crust', 'shell', 'wxrc']
+        files = []
+        dirs = []
+        if LooseVersion(self.version) < LooseVersion("4"):
+            files.extend([os.path.join('bin', 'wxrc')])
+            dirs.extend(['include', 'share'])
+            py_bins.extend(['alacarte', 'alamode', 'wrap'])
+        else:
+            majver = '3.0'  # for some reason this is still 3.0 in ver 4.0.x
+            py_bins.extend(['slices', 'slicesshell'])
+
+        files.extend([os.path.join('bin', 'py%s' % x) for x in py_bins])
+        dirs.extend([self.pylibdir])
+
+        if LooseVersion(self.version) < LooseVersion("4"):
+            libfiles = ['lib%s-%s.%s' % (x, majver, shlib_ext) for x in ['wx_baseu', 'wx_gtk2u_core']]
+            files.extend([os.path.join('lib', f) for f in libfiles])
+        else:
+            if not get_software_root('wxWidgets'):
+                libfiles = ['lib%s-%s.%s' % (x, majver, shlib_ext) for x in ['wx_baseu', 'wx_gtk3u_core']]
+                files.extend([os.path.join(self.pylibdir, 'wx', f) for f in libfiles])
+
         custom_paths = {
-            'files': ['bin/wxrc'] + [os.path.join('bin', 'py%s' % x) for x in py_bins] +
-                     [os.path.join('lib/lib%s-%s.%s' % (x, majver, shlib_ext)) for x in ['wx_baseu', 'wx_gtk2u_core']],
-            'dirs': ['include', 'share', self.pylibdir],
+            'files': files,
+            'dirs': dirs,
         }
 
         # test using 'import wx' (i.e. don't use 'import wxPython')
         self.options['modulename'] = 'wx'
 
-        # also test importing wxversion
-        custom_commands = [(self.python_cmd, '-c "import wxversion"')]
+        if LooseVersion(self.version) < LooseVersion("4"):
+            # also test importing wxversion
+            custom_commands = [(self.python_cmd, '-c "import wxversion"')]
```

Comment:
this doesn't work anymore in version > 4?

Reply:
Nope, wxversion doesn't exist in 4.x